rpcs3/Emu/Cell/Modules/cellGame.cpp

```diff
@@ -461,6 +461,24 @@ error_code cellGameDataCheck(u32 type, vm::cptr<char> dirName, vm::ptr<CellGameC
 	// TODO: not sure what should be checked there
 
 	const auto perm = g_fxo->get<content_permission>();
+	auto sfo = psf::load_object(fs::file(vfs::get(dir + "/PARAM.SFO")));
+
+	if (![&]()
+	{
+		switch (type)
+		{
+		case CELL_GAME_GAMETYPE_HDD: return psf::get_string(sfo, "CATEGORY") == "HG";
+		case CELL_GAME_GAMETYPE_GAMEDATA: return psf::get_string(sfo, "CATEGORY") == "GD";
+		case CELL_GAME_GAMETYPE_DISC: return psf::get_string(sfo, "CATEGORY") == "DG";
+		default: ASSUME(0);
+		}
+	}())
+	{
+		if (fs::is_file(vfs::get(dir + "/PARAM.SFO"))
```

Comment:
why isn't this done the other way around.
Also there's a lot of duplication in this.
And why is it checked before the init check?

Reply:
OK I will reimplement stx::init_mutex so it can support spurious initialization later for this to be 100%.

Reply:
nvm fixed didnt know cancel() exists.

Reply:
The only thing left that's weird is that fsfile operations are done twice and that the sdk params are checked/loaded before checking if the file exists.
There's probably a way to write this differently to make it more logical. But that's probably just a nitpick